pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -364,36 +364,55 @@ func (m *ManagerImpl) isVersionCompatibleWithPlugin(versions []string) bool {
 	return false
 }
 
+func (m *ManagerImpl) getDevicesToReuse(podUID string) (map[string]sets.String, bool) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	devices, ok := m.devicesToReuse[podUID]
+	return devices, ok
+}
+
+func (m *ManagerImpl) seDevicesToReuse(podUID string, devices map[string]sets.String) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	m.devicesToReuse[podUID] = devices
+}
+
+func (m *ManagerImpl) clearDevicesToReuse(podUID string) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	delete(m.devicesToReuse, podUID)
+}
+
 // Allocate is the call that you can use to allocate a set of devices
 // from the registered device plugins.
 func (m *ManagerImpl) Allocate(pod *v1.Pod, container *v1.Container) error {
-	if _, ok := m.devicesToReuse[string(pod.UID)]; !ok {
-		m.devicesToReuse[string(pod.UID)] = make(map[string]sets.String)
+	if _, ok := m.getDevicesToReuse(string(pod.UID)); !ok {
+		m.seDevicesToReuse(string(pod.UID), make(map[string]sets.String))
 	}
 	// If pod entries to m.devicesToReuse other than the current pod exist, delete them.
 	for podUID := range m.devicesToReuse {
```

Comment:
Should we lock this piece of code to protect it?
```	
for podUID := range m.devicesToReuse {
		if podUID != string(pod.UID) {
			m.clearDevicesToReuse(podUID)
		}
	}
```

Reply:
I refactor it.